* Tools

Some tools that I wrote.

** pileupcvg.c

Take as input a bam file and a chromosome id and outputs the coverage at each
nucleotide position in the chromosome.

** resample.c

Reads a table from standard input and resamples it into bins.

** average.py

Reads table from standard input and calculates average (mean, median or mode)

** bambaseatpos.c

Prints in SAM format those alignments from a BAM file which contain a given
base (or indel) at a given reference position
